libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -353,9 +353,8 @@ void DataFlowAnalyzer::clearValues(set<YulString> _variables)
 
 	// Also clear variables that reference variables to be cleared.
 	for (auto const& variableToClear: _variables)
-		for (auto const& [ref, names]: m_state.references)
-			if (names.count(variableToClear))
-				_variables.emplace(ref);
+		if (auto&& references = m_state.references.getReversedOrNullptr(variableToClear))
+			_variables += *references;
```

Comment:
It's handled already (see `if` condition). What do you mean by having two things named `references`? `auto&& references` and `m_state.references`?

Reply:
> It's handled already (see `if` condition).

Ah, you're right. That's why I'm not really sold on this if-declaration syntax too much. It can be convenient but sometimes it just does not register as a proper condition when I read it :)

> What do you mean by having two things named `references`? `auto&& references` and `m_state.references`?

Yes.